drivers/modem/ublox-sara-r4.c

```diff
@@ -421,6 +421,110 @@ MODEM_CMD_DEFINE(on_cmd_atcmdinfo_rssi_csq)
 }
 #endif
 
+/** Handler: +UULOC:
+ * <date>[1](10),<time>[2](12),<lat>[3](10),
+ * <long>[4](10),<alt>[5](10?),<uncertainty>[6](8)
+ *       ONLY for <response_type>=0:
+ * example +UULOC: 13/04/2011,09:54:51.000,45.6334520,-1.5420349,49,1  **/
+static void on_cmd_sockread_localization(struct net_buf **buf, u16_t len)
+{
+	struct net_buf *frag = NULL;
+	u16_t offset;
+
+	int i = 0, ii = 0, param_count = 0;
+	s32_t loc_data;
+	size_t value_size, out_len;
+	char value[80];
+
+	value_size = sizeof(value);
+
+	/* make sure IMEI data is received */
```

Comment:
This looks weird.  Is there a guarantee that waiting 50 ms will produce a complete buffer?  Does this need to be a loop, or at least have a post-condition check to confirm enough data is present?

Reply:
Part of the issue is this only cover the response_type 0 of which there are several option.
The response is based on location config (AT+ULOC=<mode>,<sensor>, <response_type>,<timeout>, <accuracy>[,<num_hypothesis>])

The delay is to ensure filling the buffer that could be up to, the base option 0 returns 88 characters, including all the parameters in the data struct. Its the age old until I read the data i wont know how much data to expect.

I'm sure it could be proved